lib/DDG/Spice/PublicHolidays.pm

```diff
@@ -0,0 +1,121 @@
+package DDG::Spice::PublicHolidays;
+
+# ABSTRACT: Provides the list of public holidays in a calendar year for a country. 
+
+use strict;
+use DDG::Spice;
+use Locale::Country;
+
+spice is_cached => 1;
+spice proxy_cache_valid => "200 30d";
+spice wrap_jsonp_callback => 0;
+
+spice from => "(.+)/(.+)";
+spice to => 'https://api.xmltime.com/holidays?accesskey={{ENV{DDG_SPICE_TIME_AND_DATE_ACCESSKEY}}}&secretkey={{ENV{DDG_SPICE_TIME_AND_DATE_SECRETKEY}}}&version=2&callback={{callback}}&types=federal&country=$1&year=$2';
+
+my @triggers  = ('public holidays', 'national holidays', 'bank holidays', 'federal holidays');
+my $triggers  = join('|', @triggers);
+my $countries = join('|', share("countries.txt")->slurp(chomp => 1));
+
+triggers any => @triggers;
+    
+handle query_lc => sub {
+    my $query = $_;
+    
+    # Current year and users location are the defaults unless otherwise specified by the query
+    my $defaultYear = (localtime(time))[5] + 1900;
+    my $defaultCountry = $loc->country_name;
+    
+    # Regexes to match components of queries relevant to this IA
+    my $holidays = qr/$triggers/;
+    my $country  = qr/$countries/;
+    my $year     = qr/[1-9]{1}[0-9]{3}/;
+    
+    # Regexes to ignore optional words that can precede the year or country name
+    my $in       = qr/(?:in )?/;
+    my $inThe    = qr/(?:in the |in )?/;
+    
+    my $chosenYear;
+    my $chosenCountry;
+    
+    # Match: <trigger>
```

Comment:
@moollaza 

I can have a look at simplifying things in the way you suggested but I'm concious of over-triggering. With the approach you suggested whenever a trigger, country and year appear in the query this IA will fire, regardless of the composition of the query or any extra words which may alter the meaning. My approach thus far was to strictly only respond when I could be certain the query entered was relevant.

There's also some edge cases that need more consideration. Perhaps we could talk more offline?